rundeckapp/grails-app/init/rundeckapp/init/prebootstrap/UpgradeH2DatabasePreBootstrap.groovy

```diff
@@ -0,0 +1,173 @@
+                return false
+            }
+        }
+        if(traceDbBackup.exists()) {
+            if(traceDb.exists()) {
+                traceDbBackup.delete()
+            } else {
+                logger.warn("There is a backup file but the original database does not exist. Please verify that the database has been removed and, if so, restore it before proceeding.")
+                return false
+            }
+        }
+
+        Files.copy(db.toPath(), dbBackup.toPath())
+        Files.copy(traceDb.toPath(), traceDbBackup.toPath())
+
+        logger.info("Backup created")
+
+        return dbBackup.exists() && traceDbBackup.exists()
+    }
+
+    private Properties loadConfigProperties(){
+        CoreConfigurationPropertiesLoader rundeckConfigPropertyFileLoader = new DefaultRundeckConfigPropertyLoader()
+        ServiceLoader<CoreConfigurationPropertiesLoader> rundeckPropertyLoaders = ServiceLoader.load(
+                CoreConfigurationPropertiesLoader
+        )
+        rundeckPropertyLoaders.each { loader ->
+            rundeckConfigPropertyFileLoader = loader
+        }
+        rundeckConfigPropertyFileLoader.loadProperties()
+    }
+
+    private void importToNewDatabaseVersion(String scriptFileName){
+        logger.info("Importing data from script file")
+
+        Map cfg = [:]
+        def loadedProperties = loadConfigProperties()
+        cfg.url = loadedProperties.getProperty("dataSource.url")
+        cfg.driverClassName = loadedProperties.
+                getProperty("dataSource.driverClassName", "org.h2.Driver")
+        cfg.userName = loadedProperties.getProperty("dataSource.username", "sa")
+        cfg.pwd = loadedProperties.getProperty("dataSource.password", "")
+        HikariDataSource hds
+        try {
+
+            hds = new HikariDataSource(new HikariConfig(jdbcUrl: cfg.url, username: cfg.userName,
+                    password: cfg.pwd, driverClassName: cfg.driverClassName, maxPoolSize: 2, idleTimeout: 10000L,
+                    initializationFailTimeout: TimeUnit.SECONDS.toMillis(120)))
+
+            logger.info("Importing data by SQL command: RUNSCRIPT FROM '" + scriptFileName + "' FROM_1X")
+            Sql sql = new Sql(hds)
+            sql.execute("RUNSCRIPT FROM '" + scriptFileName + "' FROM_1X")
+        } catch (Exception e){
+            logger.error("Error to import from script file: " + e.getMessage())
+        } finally {
+            logger.info("Database imported succesfully")
+            if(hds) {
+                hds.close()
+            }
+        }
+    }
+
+    private boolean exportScriptOldDatabase(){
+        logger.info("Exporting database to a script file")
+
+        boolean result = false
+        try {
+
+            def cmd = ['/bin/sh', '-c', "java -cp " + libPath + " org.h2.tools.Script -url 'jdbc:h2:file:" + dbPath + ";DB_CLOSE_ON_EXIT=FALSE' -user 'sa' -password ''"]
```

Comment:
Username and password should not be hardcoded. Like dbUrl, the value of these two variables can be extracted from the configurations.